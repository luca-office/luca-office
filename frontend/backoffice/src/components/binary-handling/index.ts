export * from "./binary-form-placeholder/binary-form-placeholder"
export * from "./binary-update-modal/binary-entry"
export * from "./binary-update-modal/binary-update-modal"
export * from "./detail-view-binary/detail-view-binary"
export * from "./image-upload-modal/image-upload-modal"
