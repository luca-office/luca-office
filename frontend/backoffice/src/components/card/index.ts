// special backoffice card elements
export * from "./creation-card"
export * from "./duration-info/card-duration-info"
export * from "./tabbed-card/__mocks__/card-tabs.mock"
export * from "./tabbed-card/tabbed-card"
