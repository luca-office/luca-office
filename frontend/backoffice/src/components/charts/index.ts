export * from "./donut-chart/donut-chart-with-legend"
