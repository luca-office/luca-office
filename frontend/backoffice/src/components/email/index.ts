export * from "./email-body-meta-entry/email-body-meta-entry"
export * from "./email-body-setting-selection/email-body-setting-selection"
export * from "./email-content/email-content"
export * from "./email-directories-list/email-directories-list"
export * from "./email-header/email-header"
export * from "./email-unread-indicator/email-unread-indicator"
