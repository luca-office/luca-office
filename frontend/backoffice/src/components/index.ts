export * from "./binary-handling"
export * from "./card"
export * from "./card-overview/card-overview"
export * from "./charts"
export * from "./creation-bar/creation-bar"
export * from "./creation-status/creation-status"
export * from "./editable-multiline-textarea/editable-multiline-textarea"
export * from "./editing-status-indicator/editing-status-indicator"
export * from "./email"
export * from "./information-entry/information-entry"
export * from "./inline-editable-header/inline-editable-header"
export * from "./inline-editable-textarea/inline-editable-textarea"
export * from "./markdown-text/markdown-text"
export * from "./markdown-text-field/markdown-text-field"
export * from "./markdown-text-update-modal/markdown-text-update-modal"
export * from "./meta-entry/meta-entry"
export * from "./meta-entry-tags/meta-entry-tags"
export * from "./overlay-edit-field/overlay-edit-field"
export * from "./rating-progress-card/rating-progress-card"
export * from "./rating-progress-card/utils"
export * from "./resort-modal/resort-modal"
export * from "./sub-header-detail/sub-header-detail"
export * from "./sub-header-detail/sub-header-detail-container"
export * from "./sub-header-filter/sub-header-filter"
export * from "./table-action-bar/table-action-bar"
export * from "./upload-file-modal/upload-file-modal"
