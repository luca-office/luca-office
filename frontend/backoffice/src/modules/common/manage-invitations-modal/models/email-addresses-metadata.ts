export interface EmailAddressesMetadata {
  readonly validAddresses: string[]
  readonly uniqueAddresses: string[]
  readonly hasInvalidAddress: boolean
  readonly duplicatedAddresses: string[]
  readonly alreadyInvitedAddresses: string[]
}
