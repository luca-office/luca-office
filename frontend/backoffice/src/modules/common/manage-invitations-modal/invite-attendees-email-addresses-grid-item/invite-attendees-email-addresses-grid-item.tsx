import {css} from "@emotion/react"
import * as React from "react"
import {Icon, Paper, Text, Tooltip} from "shared/components"
import {IconName} from "shared/enums"
import {errorColor, Flex, spacing, spacingCard, spacingMedium, spacingTiny, TextSize} from "shared/styles"
import {LucaI18nLangKey, useLucaTranslation} from "shared/translations"

export interface InviteAttendeesEmailAddressesGridItemProps {
  readonly emailAddress: string
  readonly isEmpty: boolean
  readonly disabled?: boolean
  readonly isInvalid?: boolean
  readonly onDeleteClicked?: () => void
  readonly titleKey?: LucaI18nLangKey
}

export const InviteAttendeesEmailAddressesGridItem: React.FC<InviteAttendeesEmailAddressesGridItemProps> = ({
  emailAddress,
  isEmpty,
  isInvalid = false,
  onDeleteClicked,
  disabled = false,
  titleKey
}) => {
  const {t} = useLucaTranslation()

  const paperElem = (
    <Paper
      title={disabled ? t("projects__survey_details_tooltip_invite_already_sent") : titleKey ? t(titleKey) : undefined}
      customStyles={styles.paper(isEmpty, disabled, isInvalid)}>
      <Text size={TextSize.Medium}>{emailAddress}</Text>
      {!isEmpty && (
        <Tooltip title={t("projects__survey_details_tooltip_remove_from_list")}>
          <Icon
            title={t("projects__survey_details_tooltip_remove_from_list")}
            onClick={onDeleteClicked}
            className="trash-icon"
            customStyles={styles.icon}
            name={IconName.Trash}
          />
        </Tooltip>
      )}
    </Paper>
  )

  // only display the tooltip if the paper is disabled
  return !disabled && !titleKey ? (
    paperElem
  ) : (
    <Tooltip title={t(titleKey ?? "projects__survey_details_tooltip_invite_already_sent")}>{paperElem}</Tooltip>
  )
}

const styles = {
  paper: (isEmpty: boolean, isDisabled: boolean, isInvalid: boolean) =>
    css(Flex.row, {
      padding: isEmpty ? spacingMedium : spacing(spacingTiny * 1.5, spacingCard),
      backgroundColor: isEmpty ? "#F5F6F7" : "white",
      border: isInvalid ? `1px solid ${errorColor}` : undefined,
      opacity: isDisabled ? 0.48 : 1,
      boxShadow: isEmpty ? "none" : undefined,
      cursor: isDisabled ? "not-allowed" : undefined,
      justifyContent: "space-between",

      "&:hover": {
        ".trash-icon": {
          display: !isDisabled ? "block" : "none",
          cursor: !isDisabled ? "pointer" : "initial"
        }
      }
    }),
  icon: css({
    display: "none"
  })
}
