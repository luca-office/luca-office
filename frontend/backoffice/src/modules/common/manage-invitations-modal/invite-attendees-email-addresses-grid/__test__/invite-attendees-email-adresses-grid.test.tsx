import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Heading, Icon, Text, Tooltip} from "shared/components"
import {useLucaTranslation} from "shared/translations"
import {emailAddressesMetadataMock} from "../../__mocks__/email-addresses-metadata.mock"
import {InviteAttendeesEmailAddressesGridItem} from "../../invite-attendees-email-addresses-grid-item/invite-attendees-email-addresses-grid-item"
import {
  InviteAttendeesEmailAddressesGrid,
  InviteAttendeesEmailAddressesGridProps
} from "../invite-attendees-email-addresses-grid"

const mockedAlreadyInvitedEmailAddresses = emailAddressesMetadataMock.alreadyInvitedAddresses
const mockedEmailAddresses = emailAddressesMetadataMock.validAddresses

const defaultProps: InviteAttendeesEmailAddressesGridProps = {
  emailAddressesMetadata: emailAddressesMetadataMock,
  onDeleteEmailAddress: jest.fn(),
  alreadyInvitedEmailAddresses: mockedAlreadyInvitedEmailAddresses
}

const getComponent = (props?: Partial<InviteAttendeesEmailAddressesGridProps>) => (
  <InviteAttendeesEmailAddressesGrid {...{...defaultProps, ...props}} />
)

describe("manage-invitations-modal-email-addresses-grid", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure - two mails", () => {
    const component = getComponent({
      emailAddressesMetadata: {
        ...emailAddressesMetadataMock,
        validAddresses: [mockedEmailAddresses[0], mockedEmailAddresses[1]],
        uniqueAddresses: [mockedEmailAddresses[0], mockedEmailAddresses[1]],
        duplicatedAddresses: [],
        alreadyInvitedAddresses: []
      },
      alreadyInvitedEmailAddresses: []
    })
    const tree = shallow(component)

    expect(tree.find(".wrapper")).toHaveLength(1)
    expect(tree.find(Heading)).toHaveLength(1)
    expect(tree.find(".grid-wrapper")).toHaveLength(1)
    expect(tree.find(".grid-top-bar")).toHaveLength(1)
    expect(tree.find(".grid-scroll-wrapper")).toHaveLength(1)
    expect(tree.find(InviteAttendeesEmailAddressesGridItem)).toHaveLength(2)
    tree.find(InviteAttendeesEmailAddressesGridItem).forEach(node => {
      expect(node.prop("isEmpty")).toBe(false)
    })
    expect(tree.find(InviteAttendeesEmailAddressesGridItem).first().prop("emailAddress")).toBe(mockedEmailAddresses[0])
    expect(tree.find(InviteAttendeesEmailAddressesGridItem).last().prop("emailAddress")).toBe(mockedEmailAddresses[1])
    expect(tree.find(".grid-bottom-bar")).toHaveLength(1)
  })

  it("has correct structure - empty emails", () => {
    const component = getComponent({
      emailAddressesMetadata: {
        ...emailAddressesMetadataMock,
        validAddresses: [],
        uniqueAddresses: [],
        duplicatedAddresses: [],
        alreadyInvitedAddresses: []
      },
      alreadyInvitedEmailAddresses: []
    })
    const tree = shallow(component)

    expect(tree.find(".wrapper")).toHaveLength(1)
    expect(tree.find(Heading)).toHaveLength(1)
    expect(tree.find(".grid-wrapper")).toHaveLength(1)
    expect(tree.find(".grid-top-bar")).toHaveLength(1)
    expect(tree.find(".grid-scroll-wrapper")).toHaveLength(1)
    expect(tree.find(Text)).toHaveLength(1)
    tree.find(InviteAttendeesEmailAddressesGridItem).forEach(node => {
      expect(node.prop("isEmpty")).toBe(true)
      expect(node.prop("emailAddress")).toBe("")
    })
    expect(tree.find(".grid-bottom-bar")).toHaveLength(1)
  })

  it("has correct structure - with one existing emails and no new emails", () => {
    const component = getComponent({
      emailAddressesMetadata: {
        ...emailAddressesMetadataMock,
        validAddresses: [],
        uniqueAddresses: [],
        duplicatedAddresses: [],
        alreadyInvitedAddresses: []
      },
      alreadyInvitedEmailAddresses: mockedAlreadyInvitedEmailAddresses
    })
    const tree = shallow(component)

    tree.find(InviteAttendeesEmailAddressesGridItem).forEach((node, index) => {
      expect(node.prop("isEmpty")).toBe(false)
      expect(node.prop("emailAddress")).toBe(mockedAlreadyInvitedEmailAddresses[index])
      expect(node.prop("disabled")).toBe(true)
    })
  })

  it("has correct structure - with one existing emails and two new emails", () => {
    const component = getComponent({
      emailAddressesMetadata: {
        ...emailAddressesMetadataMock,
        validAddresses: [mockedEmailAddresses[0], mockedEmailAddresses[1]],
        uniqueAddresses: [mockedEmailAddresses[0], mockedEmailAddresses[1]],
        duplicatedAddresses: [],
        alreadyInvitedAddresses: []
      },
      alreadyInvitedEmailAddresses: mockedAlreadyInvitedEmailAddresses
    })
    const tree = shallow(component)

    const addresses = [...mockedAlreadyInvitedEmailAddresses, ...[mockedEmailAddresses[0], mockedEmailAddresses[1]]]

    tree.find(InviteAttendeesEmailAddressesGridItem).forEach((node, index) => {
      expect(node.prop("isEmpty")).toBe(false)
      expect(node.prop("emailAddress")).toBe(addresses[index])
      if (index < 1) {
        // first email should be disabled, because it's already invited
        expect(node.prop("disabled")).toBe(true)
      } else {
        expect(node.prop("disabled")).toBe(false)
      }
    })
  })

  it("displays the tooltip on hover of trashcan icon", () => {
    const component = getComponent({
      emailAddressesMetadata: {...emailAddressesMetadataMock, uniqueAddresses: mockedEmailAddresses},
      alreadyInvitedEmailAddresses: []
    })
    const tree = shallow(component)
    const {t} = useLucaTranslation()

    const gridItem = tree.find(InviteAttendeesEmailAddressesGridItem).first().dive()
    const trashcan = gridItem.find(Icon)
    trashcan.simulate("mouseover")
    expect(gridItem.find(Tooltip).first().prop("title")).toEqual(t("projects__survey_details_tooltip_remove_from_list"))
  })

  it("displays the tooltip on hover disabled grid item", () => {
    const component = getComponent({
      emailAddressesMetadata: {...emailAddressesMetadataMock, uniqueAddresses: []},
      alreadyInvitedEmailAddresses: mockedAlreadyInvitedEmailAddresses
    })
    const tree = shallow(component)
    const {t} = useLucaTranslation()

    const gridItem = tree.find(InviteAttendeesEmailAddressesGridItem).first().dive()
    gridItem.simulate("mouseover")
    expect(gridItem.find(Tooltip).first().prop("title")).toEqual(
      t("projects__survey_details_tooltip_invite_already_sent")
    )
  })
})
