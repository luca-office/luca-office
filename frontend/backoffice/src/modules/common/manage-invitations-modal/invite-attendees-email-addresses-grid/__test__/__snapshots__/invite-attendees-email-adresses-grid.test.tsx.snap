// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manage-invitations-modal-email-addresses-grid renders correctly 1`] = `
<div
  className="wrapper css-z70zj5-InviteAttendeesEmailAddressesGrid"
>
  <div
    className="css-1akphf9-Heading"
  >
    projects__survey_details_invite_emails_grid_heading
  </div>
  <div
    className="grid-wrapper css-1d467hy-InviteAttendeesEmailAddressesGrid"
  >
    <div
      className="grid-top-bar css-7h2thf-InviteAttendeesEmailAddressesGrid"
    />
    <div
      className="grid-scroll-wrapper css-v0ses7-InviteAttendeesEmailAddressesGrid"
    >
      <div
        className="grid css-b7vqx3-InviteAttendeesEmailAddressesGrid"
      >
        <div
          className="css-0"
          id="tooltip_42"
        >
          <div>
            <div
              className="css-bn1a4g-Paper"
              title="projects__survey_details_tooltip_invite_already_sent"
            >
              <div
                className="css-17zbm7y-Text"
              >
                cap@cap3.de
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <div
                  className="trash-icon css-16kls7j-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                  onClick={[Function]}
                  title="projects__survey_details_tooltip_remove_from_list"
                />
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                projects__survey_details_tooltip_remove_from_list
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
          role="tooltip"
          style={
            {
              "backgroundColor": "#2e3032",
              "zIndex": 1,
            }
          }
        >
          projects__survey_details_tooltip_invite_already_sent
          <div
            className="react-tooltip-arrow styles-module_arrow__K0L3T"
            style={{}}
          />
        </div>
        <div>
          <div
            className="css-ujfly3-Paper"
          >
            <div
              className="css-17zbm7y-Text"
            >
              testmail@test.de
            </div>
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="trash-icon css-16kls7j-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
                title="projects__survey_details_tooltip_remove_from_list"
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__survey_details_tooltip_remove_from_list
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="css-ujfly3-Paper"
          >
            <div
              className="css-17zbm7y-Text"
            >
              cap@cap3.de
            </div>
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="trash-icon css-16kls7j-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
                title="projects__survey_details_tooltip_remove_from_list"
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__survey_details_tooltip_remove_from_list
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="css-ujfly3-Paper"
          >
            <div
              className="css-17zbm7y-Text"
            >
              testmai2l@test.de
            </div>
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="trash-icon css-16kls7j-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
                title="projects__survey_details_tooltip_remove_from_list"
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__survey_details_tooltip_remove_from_list
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="css-ujfly3-Paper"
          >
            <div
              className="css-17zbm7y-Text"
            >
              cap3@cap3.de
            </div>
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="trash-icon css-16kls7j-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
                title="projects__survey_details_tooltip_remove_from_list"
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__survey_details_tooltip_remove_from_list
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
        </div>
        <div
          className="css-0"
          id="tooltip_42"
        >
          <div>
            <div
              className="css-7t4i1z-Paper"
              title="projects__survey_details_tooltip_invite_duplicate"
            >
              <div
                className="css-17zbm7y-Text"
              >
                testmail@test.de
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <div
                  className="trash-icon css-16kls7j-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                  onClick={[Function]}
                  title="projects__survey_details_tooltip_remove_from_list"
                />
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                projects__survey_details_tooltip_remove_from_list
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
          role="tooltip"
          style={
            {
              "backgroundColor": "#2e3032",
              "zIndex": 1,
            }
          }
        >
          projects__survey_details_tooltip_invite_duplicate
          <div
            className="react-tooltip-arrow styles-module_arrow__K0L3T"
            style={{}}
          />
        </div>
        <div
          className="css-0"
          id="tooltip_42"
        >
          <div>
            <div
              className="css-7t4i1z-Paper"
              title="projects__survey_details_tooltip_invite_already_sent"
            >
              <div
                className="css-17zbm7y-Text"
              >
                cap@cap3.de
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <div
                  className="trash-icon css-16kls7j-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                  onClick={[Function]}
                  title="projects__survey_details_tooltip_remove_from_list"
                />
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                projects__survey_details_tooltip_remove_from_list
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
          role="tooltip"
          style={
            {
              "backgroundColor": "#2e3032",
              "zIndex": 1,
            }
          }
        >
          projects__survey_details_tooltip_invite_already_sent
          <div
            className="react-tooltip-arrow styles-module_arrow__K0L3T"
            style={{}}
          />
        </div>
      </div>
    </div>
    <div
      className="grid-bottom-bar css-121255i-InviteAttendeesEmailAddressesGrid"
    />
  </div>
</div>
`;
